### ISSUE TITLE
Deal with the optional use of GROUP BY in query-command.js: selectCommandDetailsFromAllEndpointTypesAndClusters

### ISSUE THREAD

**Issue by tecimovic** [project-chip/zap#192]
There was an issue with user_cluster_has_enabled_command helper. It is returning non-deterministic results because the GROUP BY in the query queryCommand.selectCommandDetailsFromAllEndpointTypesAndClusters() has a GROUP BY but no aggregates, so essentially it's simply picking the first value from the rows which have duplicates.

Given that the sorting order there is irrelevant, we should not be doing that. I suspect this query comes from assumption that the configuration of a command across multiple endpoints MUST be the same (as is the case in Silabs ZIgbee Pro implementation).

In case of Matter, they may be different, so we need to fix this.

I've temporarily patched the matter by making GROUP BY optional in the query, that but that just patches the problem for now.

**Comment by tecimovic** [comment]
@brdandu : there is the same problem with the following queries as well:

query-command.js:
  - selectAllAvailableClusterCommandDetailsFromEndpointTypes
  - selectAllClustersWithIncomingCommands
  - selectAllIncomingCommandsForCluster
  - selectAllCommandDetailsFromEnabledClusters
  - selectAllCliCommandDetailsFromEnabledClusters
  - selectCommandDetailsFromAllEndpointTypesAndClusters
  - selectCommandDetailsFromAllEndpointTypesAndClustersUtil

query-endpoint-type.js:
  - selectAllClustersDetailsFromEndpointTypes
  - selectEndpointDetailsFromAddedEndpoints
  - selectAllClustersNamesFromEndpointTypes
  - selectAllClustersDetailsIrrespectiveOfSideFromEndpointTypes

query-cluster.js:
  - selectClusterDetailsFromEnabledClusters

query-attribute.js:
  - selectAllAttributeDetailsFromEnabledClusters
  - selectAttributeDetailsFromAllEndpointTypesAndClustersUtil
  - selectAttributeDetailsWithABoundFromEnabledClusters
  - selectAttributeDetailsFromEnabledClusters
  - selectAttributeBoundDetails
  - selectReportableAttributeDetailsFromEnabledClustersAndEndpoints

Bottom line with these are:
   whenever you are doing GROUP BY, and at the same time the query is NOT doing a "HAVING" statement or having some kind of aggregate functions (such as COUNT, MIN, MAX, AVG, etc...) that operate on the group range, the query is suspect of non-determinism or favoritism (randomly favoring one result row over the other, based on random resorting).

So we need to take a look and understand these. I think it's very possible that this stems from the fact that embedded code in ZigbeePro SDK does not support the concept of same clusters on different endpoints having different configurations. But that is a specific case for ZigbeePro. We can't put this limitation in Zap overall.
